pytensor/link/jax/dispatch/random.py

```diff
@@ -450,6 +443,42 @@ def sample_fn(rng, size, dtype, n, p):
     return sample_fn
 
 
+def _jax_multinomial(n, p, shape=None, key=None):
+    if jnp.shape(n) != jnp.shape(p)[:-1]:
+        broadcast_shape = jax.lax.broadcast_shapes(jnp.shape(n), jnp.shape(p)[:-1])
+        n = jnp.broadcast_to(n, broadcast_shape)
+        p = jnp.broadcast_to(p, broadcast_shape + jnp.shape(p)[-1:])
+
+    if shape is not None:
+        broadcast_shape = jax.lax.broadcast_shapes(jnp.shape(n), shape)
+        n = jnp.broadcast_to(n, broadcast_shape)
+
+    else:
+        shape = p.shape[:-1]
+
+    p = p / jnp.sum(p, axis=-1, keepdims=True)
```

Comment:
We should assume p is valid, and not renormalize for the user. That would mask errors